File: 442_Find_All_Duplicates_in_an_Array.js
var findDuplicates = function(nums) {
    let ans = []
    let t = 0
    for(let i = 0; i < nums.length; i++){
        if(nums[i] < 0) t = Math.abs(nums[i]) - 1
        else t = nums[i] - 1
        if(nums[t] > 0) nums[t] = -nums[t]
        else ans.push(Math.abs(nums[i]))
    }
    return ans
};

File: 442_Find_All_Duplicates_in_an_Array.cpp
class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) {
        vector<int> ans;
        int t = 0;
        for(int i = 0; i < nums.size(); i++){
            if(nums[i] < 0) t = abs(nums[i]) - 1;
            else t = nums[i] - 1;
            if(nums[t] > 0) nums[t] = -nums[t];
            else ans.push_back(abs(nums[i]));
        }
        return ans;
    }
};
